packages/web-components/src/components/back-to-top/back-to-top.ts

```diff
@@ -90,17 +102,28 @@ class DDSBackToTop extends HostListenerMixin(StableSelectorMixin(LitElement)) {
     }
   }
 
-  private _observeResizeBody = (entries) => {
+  /**
+   * ResizeObserver callback function.
+   *
+   * We only observe the body element, therefore entries will typically only
+   * have a single item corresponding to changes in the size of the body
+   * element.
+   */
+  private _observeResizeBody = (entries: ResizeObserverEntry[]) => {
     this._bodyHeight = entries[entries.length - 1].target.scrollHeight;
+    // Body height changes may require adjustment to the current visibility
+    // of the back to top button.
+    this._manageVisibility();
   };
 
   /**
-   * Show button only when document height is 3x greater than viewport
+   * The back to top button is only necessary when the document height
+   * is 3x greater than the viewport.
    */
-  private _showBackToTop() {
+  private _isBackToTopNeededForPage() {
     // @ts-ignore
-    this._windowHeight = this.ownerDocument.defaultView?.innerHeight;
-    return this._bodyHeight > this._windowHeight * 3;
+    this._viewportHeight = this.ownerDocument.defaultView?.innerHeight;
```

Comment:
TIL: [Document.defaultView](https://developer.mozilla.org/en-US/docs/Web/API/Document/defaultView)

Reply:
MDN and I were great friends through this, for me also:

* [ownerDocument](https://developer.mozilla.org/en-US/docs/Web/API/Node/ownerDocument)
* [scrollingElement](https://developer.mozilla.org/en-US/docs/Web/API/Document/scrollingElement)